system_tests/retryable_test.go

```diff
@@ -579,3 +633,240 @@ func TestL1FundedUnsignedTransaction(t *testing.T) {
 		t.Errorf("L2 transaction: %v has failed", receipt.TxHash)
 	}
 }
+
+func TestRetryableSubmissionAndRedeemFees(t *testing.T) {
+	l2info, l1info, l2client, l1client, delayedInbox, lookupL2Hash, ctx, teardown := retryableSetup(t)
+	defer teardown()
+	infraFeeAddr, networkFeeAddr := setupFeeAddresses(t, ctx, l2client, l2info)
+
+	ownerTxOpts := l2info.GetDefaultTransactOpts("Owner", ctx)
+	simpleAddr, simple := deploySimple(t, ctx, ownerTxOpts, l2client)
+	simpleABI, err := mocksgen.SimpleMetaData.GetAbi()
+	Require(t, err)
+
+	elevateL2Basefee(t, ctx, l2client, l2info)
+
+	infraBalanceBefore, err := l2client.BalanceAt(ctx, infraFeeAddr, nil)
+	Require(t, err)
+	networkBalanceBefore, err := l2client.BalanceAt(ctx, networkFeeAddr, nil)
+	Require(t, err)
+
+	beneficiaryAddress := l2info.GetAddress("Beneficiary")
+	deposit := arbmath.BigMul(big.NewInt(1e12), big.NewInt(1e12))
+	callValue := common.Big0
+	usertxoptsL1 := l1info.GetDefaultTransactOpts("Faucet", ctx)
+	usertxoptsL1.Value = deposit
+	baseFee := GetBaseFee(t, l2client, ctx)
+	l1tx, err := delayedInbox.CreateRetryableTicket(
+		&usertxoptsL1,
+		simpleAddr,
+		callValue,
+		big.NewInt(1e16),
+		beneficiaryAddress,
+		beneficiaryAddress,
+		// send enough L2 gas for intrinsic but not compute
+		big.NewInt(int64(params.TxGas+params.TxDataNonZeroGasEIP2028*4)),
+		big.NewInt(baseFee.Int64()*2),
+		simpleABI.Methods["incrementRedeem"].ID,
+	)
+	Require(t, err)
+	l1receipt, err := EnsureTxSucceeded(ctx, l1client, l1tx)
+	Require(t, err)
+	if l1receipt.Status != types.ReceiptStatusSuccessful {
+		Fatal(t, "l1receipt indicated failure")
+	}
+
+	waitForL1DelayBlocks(t, ctx, l1client, l1info)
+
+	submissionTxHash := lookupL2Hash(l1receipt)
+	submissionReceipt, err := WaitForTx(ctx, l2client, submissionTxHash, time.Second*5)
+	Require(t, err)
+	if submissionReceipt.Status != types.ReceiptStatusSuccessful {
```

Comment:
nit: EnsureTxSucceeded should work here too (and a couple more times in this function)

Reply:
I fixed it in places where we can have a pointer to `types.Transaction`, but I left `WaitForTx` in places where we get transaction hash from logs.